rfcs/0017-tx-valid-since/0017-tx-valid-since.md

```diff
@@ -15,97 +15,244 @@ This RFC suggests adding a new consensus rule to prevent a cell to be spent befo
 
 ## Summary 
 
-Transaction adds a new `u64` type field `valid_since`, which prevents the transaction to be mined before a certain point, the highest bit of `valid_since` is `type flag`, the other 63 bits represent the actual value `V`, the consensus to validate this field described as follow:
+Transaction input adds a new `u64` type field `valid_since`, which prevents the transaction to be mined before an absolute or relative time.
 
-* ignore this validate rule if all bits of `valid_since` are 0.
-* `V` represents a block number when `type flag` is 0, the validation MUST fail if `tip.block_number < V`.
-* `V` represents a block timestamp when `type flag` is 1, the timestamp is represented as `V * 512` seconds, the validation MUST fail if `tip.timestamp < V * 512`.
+The highest 8 bits of `valid_since` is `flags`, the other `56` bits represent `value`, `flags` allow us to determine behaviours:
+* `flags & (1 << 7)` represent `absolute_or_relative_type`.
+* `flags & (1 << 6)` represent `time_type`.
+* other 6 `flags` bits remain for other use.
 
-Otherwise, the transaction validation SHOULD continue.
+The consensus to validate this field described as follow:
+* iterate inputs, and validate each input by following rules.
+* ignore this validate rule if all 64 bits of `valid_since` are 0.
+* check `time_type` flag:
+    * the lower 56 bits of `valid_since` represent block number if `time_type` is `0`.
+    * the lower 56 bits of `valid_since` represent block timestamp if `time_type` is `1` (because block timestamp is low accuracy,we represent the timestamp as `(valid_since & 0x00ffffffffffffff) << 9` seconds to save space).
```

Comment:
I don't get the point here: why `<< 9`? What's the time unit for `valid_since` here?

Reply:
`512 seconds` == `1 << 9`

Reply:
`512` seconds is not suitable for ckb, because ckb block time could be in seconds instead of 10 minutes.

I suggest to use a seconds-level timestamp.

Reply:
Ok, make sense.